src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/CO2Solubility.cpp

```diff
@@ -265,22 +270,27 @@ TableFunction const * makeVapourisationTable( string_array const & inputParams,
                                               string const & functionName,
                                               FunctionManager & functionManager )
 {
-  // initialize the (p,T) coordinates
-  PTTableCoordinates tableCoords;
-  PVTFunctionHelpers::initializePropertyTable( inputParams, tableCoords );
-
-  // Currently initialise to all zeros
-
-  array1d< real64 > values( tableCoords.nPressures() * tableCoords.nTemperatures() );
-  values.zero();
-
   string const tableName = functionName + "_waterVaporization_table";
+
   if( functionManager.hasGroup< TableFunction >( tableName ) )
   {
-    return functionManager.getGroupPointer< TableFunction >( tableName );
+    TableFunction * const vapourisationTable = functionManager.getGroupPointer< TableFunction >( tableName );
+    vapourisationTable->initializeFunction();
+    vapourisationTable->setDimUnits( { units::Pressure, units::TemperatureInC } );
+    vapourisationTable->setValueUnits( units::Solubility );
+    vapourisationTable->setInterpolationMethod( TableFunction::InterpolationType::Linear );
```

Comment:
I would imagine we wouldn't want to override the user specified interpolation method. Same for all the other tables.

Reply:
just though being consistent with internal table interpolation is better, but you have a point, let me drop `setInterpolationMethod`